src/Events/Admin/Onboarding/Controller.php

```diff
@@ -88,6 +88,59 @@ public function handle_onboarding_page_dismiss(): void {
 		$this->container->make( Landing_Page::class )->handle_onboarding_page_dismiss();
 	}
 
+	/**
+	 * Redirects users to the Guided Setup page when accessing any TEC settings or management page for the first time.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function redirect_tec_pages_to_guided_setup(): void {
+		// Do not redirect if they are already on the Guided Setup page.
+		$page = tec_get_request_var( 'page' );
+		if ( Landing_Page::$slug === $page ) {
+			return;
+		}
+
+		// Do not redirect if they have been to the Guided Setup page already.
+		if ( (bool) tribe_get_option( 'tec_onboarding_wizard_visited_guided_setup', false ) ) {
+			return;
+		}
+
+		// Do not redirect if they dismissed the Guided Setup page.
+		if ( (bool) Landing_Page::is_dismissed() ) {
+			return;
+		}
+
+		// Do not redirect if the target is not The Events Calendar-related admin pages.
+		$post_types = apply_filters( 'tec_events_admin_post_types', [ 'tribe_events', 'tribe_event_series', 'tribe_venue', 'tribe_organizer' ] );
+
+		$post_type = tec_get_request_var( 'post_type' );
+
+		if ( ! in_array( $post_type, $post_types, true ) ) {
+			return;
+		}
```

Comment:
This is lowest cost from quering DB - Should be placed right after the page check.